File: Interfaces/ISidebarmenuLineService.cs
using WMS_WEBAPI.DTOs;
using WMS_WEBAPI.Models;

namespace WMS_WEBAPI.Interfaces
{
    public interface ISidebarmenuLineService
    {
        Task<ApiResponse<IEnumerable<SidebarmenuLineDto>>> GetAllAsync();
        Task<ApiResponse<SidebarmenuLineDto>> GetByIdAsync(long id);
        Task<ApiResponse<SidebarmenuLineDto>> CreateAsync(CreateSidebarmenuLineDto createDto);
        Task<ApiResponse<SidebarmenuLineDto>> UpdateAsync(long id, UpdateSidebarmenuLineDto updateDto);
        Task<ApiResponse<bool>> SoftDeleteAsync(long id);
        Task<ApiResponse<bool>> ExistsAsync(long id);
        Task<ApiResponse<IEnumerable<SidebarmenuLineDto>>> GetByHeaderIdAsync(int headerId);
        Task<ApiResponse<SidebarmenuLineDto>> GetByPageAsync(string page);
    }
}

File: WebApi/Interfaces/PlatformSidebarInterfaces/ISidebarmenuLineService.ts
import type { CreateSidebarmenuLineDto, SidebarmenuLineDto, UpdateSidebarmenuLineDto } from '../../Models/index';
import type { ApiResponse, PagedResponse } from '../../Models/ApiResponse';

export interface ISidebarmenuLineService {
  getAll(): Promise<ApiResponse<SidebarmenuLineDto[]>>;
  getPaged(pageNumber: number, pageSize: number, sortBy: string, sortDirection: string): Promise<ApiResponse<PagedResponse<SidebarmenuLineDto>>>;
  getById(id: number): Promise<ApiResponse<SidebarmenuLineDto>>;
  create(createDto: CreateSidebarmenuLineDto): Promise<ApiResponse<SidebarmenuLineDto>>;
  update(id: number, updateDto: UpdateSidebarmenuLineDto): Promise<ApiResponse<SidebarmenuLineDto>>;
  softDelete(id: number): Promise<ApiResponse<boolean>>;
  exists(id: number): Promise<ApiResponse<boolean>>;
  getByHeaderId(headerId: number): Promise<ApiResponse<SidebarmenuLineDto[]>>;
  getByPage(page: string): Promise<ApiResponse<SidebarmenuLineDto>>;
}
